internal/ls/completions.go

```diff
@@ -4860,3 +4985,625 @@ func createCompletionDetailsForSymbol(
+			)
+			var snippetText string
+			if isSnippet {
+				snippetText = getJSDocParamAnnotation(
+					paramName,
+					param.Initializer(),
+					param.AsParameterDeclaration().DotDotDotToken,
+					isJS,
+					/*isObject*/ false,
+					/*isSnippet*/ true,
+					typeChecker,
+					options,
+					preferences,
+					&tabstopCounter,
+				)
+			}
+			if tagNameOnly { // Remove `@`
+				displayText = displayText[1:]
+				if snippetText != "" {
+					snippetText = snippetText[1:]
+				}
+			}
+
+			return &lsproto.CompletionItem{
+				Label:            displayText,
+				Kind:             ptrTo(lsproto.CompletionItemKindVariable),
+				SortText:         ptrTo(string(SortTextLocationPriority)),
+				InsertText:       strPtrTo(snippetText),
+				InsertTextFormat: core.IfElse(isSnippet, ptrTo(lsproto.InsertTextFormatSnippet), nil),
+			}
+		} else if paramIndex == paramTagCount {
+			// Destructuring parameter; do it positionally
+			paramPath := fmt.Sprintf("param%d", paramIndex)
+			displayTextResult := generateJSDocParamTagsForDestructuring(
+				paramPath,
+				param.Name(),
+				param.Initializer(),
+				param.AsParameterDeclaration().DotDotDotToken,
+				isJS,
+				/*isSnippet*/ false,
+				typeChecker,
+				options,
+				preferences,
+			)
+			var snippetText string
+			if isSnippet {
+				snippetTextResult := generateJSDocParamTagsForDestructuring(
+					paramPath,
+					param.Name(),
+					param.Initializer(),
+					param.AsParameterDeclaration().DotDotDotToken,
+					isJS,
+					/*isSnippet*/ true,
+					typeChecker,
+					options,
+					preferences,
+				)
+				snippetText = strings.Join(snippetTextResult, options.NewLine.GetNewLineCharacter()+"* ")
+			}
+			displayText := strings.Join(displayTextResult, options.NewLine.GetNewLineCharacter()+"* ")
+			if tagNameOnly { // Remove `@`
+				displayText = strings.TrimPrefix(displayText, "@")
+				snippetText = strings.TrimPrefix(snippetText, "@")
+			}
+			return &lsproto.CompletionItem{
+				Label:            displayText,
+				Kind:             ptrTo(lsproto.CompletionItemKindVariable),
+				SortText:         ptrTo(string(SortTextLocationPriority)),
+				InsertText:       strPtrTo(snippetText),
+				InsertTextFormat: core.IfElse(isSnippet, ptrTo(lsproto.InsertTextFormatSnippet), nil),
+			}
+		}
+		return nil
+	})
+}
+
+func getJSDocParamAnnotation(
+	paramName string,
+	initializer *ast.Expression,
+	dotDotDotToken *ast.TokenNode,
+	isJS bool,
+	isObject bool,
+	isSnippet bool,
+	typeChecker *checker.Checker,
+	options *core.CompilerOptions,
+	preferences *UserPreferences,
+	tabstopCounter *int,
+) string {
+	if isSnippet {
+		// !!! Debug.assertIsDefined(tabstopCounter);
+	}
+	if initializer != nil {
+		paramName = getJSDocParamNameWithInitializer(paramName, initializer)
+	}
+	if isSnippet {
+		paramName = escapeSnippetText(paramName)
+	}
+	if isJS {
+		t := "*"
+		if isObject {
+			// !!! Debug.assert(!dotDotDotToken, `Cannot annotate a rest parameter with type 'Object'.`);
+			t = "Object"
```

Comment:
wait, what? `Object` is somehow the worst type for anything in either TS or JS. In Strada it means `any` if strict is off but in Corsa it always means `Object`. I think `object` makes more sense, or `*` or `any` if `object` is nonsensical somehow.

Reply:
wait wait. I remembered. It's probably because it's preparing to emit nested object literal type notation:

```js
/** 
 * @param {Object} param0
 * @param {string} param0.a
 * @param {number} param0.b
 */
function f({ a, b }) {}
```

So it might make sense. 
It's also one more reason why `Object -> any` in Strada, even though I hate that and it's not coming back for Corsa.

Reply:
As discussed offline, this works with `object` too:

https://www.typescriptlang.org/play/?ts=6.0.0-dev.20250814&filetype=js#code/PQKhAIChwgBAHAhgJ0QW3AbwPYCMBWApgMYAuAvuEqmgAzRzXpYDOpyAlgHYDmlTdAHSIG4BCmaYuAVzS5CyfhKG4GwSADNpXMh2xdwGgBSZwiADThc4ALzgAjOHIBKLNHAfi+ltgA2hQV9sHiNEZwBudw9kQlJpZANHAGorSPIgA

So keeping `object` until someone complains.